auto_dev/utils/signals.py

```diff
@@ -0,0 +1,46 @@
+"""Signal management utilities."""
+
+import signal
+from types import FrameType
+from contextlib import contextmanager
+from collections.abc import Callable
+
+
+CallableSignalHandler = Callable[[int, FrameType], None]
```

Comment:
theres already a utils file right? 

Could add that to that file rather than creating another sibdirectory? 
Or move the utils tools in here maybe?

Reply:
Gotcha i see what you did there